### DISCUSSION TITLE
Asynchronously modifying query data

### DISCUSSION THREAD

**Discussion by seMikel** [TanStack/query#4270] [Q&A] [answered]
I have basic setup with a query that retrieves some data, let's call it a 'user', and a mutation that modifies said 'user', updating the query cache with what comes back as the response, similar to https://tanstack.com/query/v4/docs/guides/updates-from-mutation-responses.

What I want to create is a new query, that would use the 'user' data from the initial query, but add some additional information to it. One would generally use the `select` option for that, but the issue is that the additional information is retrieved asynchronously based on the 'user' data, and the additional information needs to be retrieved and applied every time the 'user' is updated.

I can't seem to come up with a way to create such a query that would achieve that result, yet behave like a completely normal reusable query.

**Comment by TkDodo** [comment, marked as answer]
Seems like you're looking for dependent queries:

https://tanstack.com/query/v4/docs/guides/dependent-queries

it even shows as an example how to "depend on a user"

**Reply by seMikel** [reply]
Almost, if I'm not mistaken, the difference between dependant queries and my scenario is that the extra data in dependent queries is based on the user id. As a result when the mutation runs or the user's data is otherwise updated, the dependent query will not refetch.

**Reply by TkDodo** [reply]
you can also have the second query depend on the user data, not only the user id

**Reply by seMikel** [reply]
Huh, I must have had some misconception that one shouldn't be using complex objects as query keys, thanks!
